# Sample Analysis
| Result Type | Primary Result | Others |
| :--  | :--      | :--   |
| Small_Variant | <br>**(1) "deepvariant.phased.vcf.gz"** <br>**(2) "deepvariant.phased.vcf.gz.tbi"** <br>**(3) "deepvariant.vcf.stats.txt"** <br>* Small variants called by DeepVariant and phased by HiPhase (with index) <br>* only contain the positions where variants<br>* bcftools stats summary statistics for small variants| **<br>(1) "deepvariant.g.vcf.gz" <br>(2) "deepvariant.g.vcf.gz.tbi"** <br>* Small variants (SNPs and INDELs < 50bp) gVCFs called by DeepVariant (with index) <br>* gvcf: contain both variant positions and non-variant (reference) positions **<br>(1) "deepvariant.roh.bed" <br>(2) "deepvariant.bcftools_roh.out"** <br>* Regions of homozygosity determined by bcftools roh using "--AF-dflt 0.4" <br>* Output of bcftools roh using "--AF-dflt 0.4" <br>* ROH: runs of homozygosity|
| Structure_Variation  | **<br>(1) "pbsv.phased.vcf.gz" <br>(2) "pbsv.phased.vcf.gz.tbi"** <br>*Structural variants called by pbsv and phased by HiPhase (with index)   |  |
| Haplotype_Phasing_Info | **<br>(1) "hiphase.blocks.tsv"** <br>* Phase block list written by HiPhase| **<br>(1) "hiphase.haplotags.tsv"** <br>* Per-read haplotag information, written by HiPhase **<br>(1) "hiphase.stats.tsv"** <br>* Phase block summary statistics written by HiPhase |
| Haplotagged_BAM | **<br>(1) "haplotagged.bam" <br>(2) "haplotagged.bam.bai"** <br>* Aligned (by pbmm2), haplotagged (by HiPhase) reads (with index) | **<br>(1) "haplotagged.mosdepth.summary.txt"** <br>* mosdepth summary of median depths per chromosome **<br>(1) "aligned.haplotagged.regions.bed.gz"** <br>* mosdepth BED of median coverage depth per 500 bp window|
| Copy_Number_Variation | **<br>(1) "vcf.gz" <br>(2) "vcf.gz.tbi"** <br>* The copy number variation calling results | **<br>(1) "copynum.bedgraph" <br>(2) "depth.bw" <br>(3) "maf.bw"** <br>* visualize CNV-scale events in IGV|
| Tandem_Repeats | **<br>(1) "aligned.haplotagged.trgt.sorted.vcf.gz" <br>(2)"aligned.haplotagged.trgt.sorted.vcf.gz.tbi"**  <br>* Tandem repeat genotypes from TRGT (with index) | **<br>(1) "aligned.haplotagged.trgt.spanning.sorted.bam" <br>(2)"aligned.haplotagged.trgt.spanning.sorted.bam.bai"** <br>* Fragments of HiFi reads spanning loci genotyped by TRGT (with index) **<br>(1) "trgt.dropouts.txt"** <br>* Regions with insufficient coverage to genotype by TRGT |
| Medically_Relevant_Genes | **<br>(1) All vcf files inside subfolder "VCFs"** <br>* Phased Variant calls for selected medically relevant genes in segmental duplications, generated by Paraphase|**<br>(1) All files inside subfolder "Realigned_BAMs"** <br>* Output json file generated by Paraphase <br>* Realigned BAM for selected medically relevant genes in segmental duplications (with index), generated by Paraphase|
| Methylation | **<br>(1) "combined.bed" <br>(2) "hap1.bed" <br>(3) "hap2.bed"** <br>* 5mCpG site methylation probability pileups generated by pb-CpG-tools |**<br>(1) "combined.bw" <br>(2) "hap1.bw" <br>(3) "hap2.bw"** <br>* 5mCpG site methylation probability pileups generated by pb-CpG-tools <br>* Can visualize in IGV|
| Quality_Control | **<br>(1) "read_length_and_quality.tsv" <br>(2) "read_length_summary.tsv" <br>(3) "read_quality_summary.tsv"** <br>* length and quality for each read (per input BAM) <br>* For each input BAM, read length distribution (per input BAM) <br>* For each input BAM, read quality distribution (per input BAM)| |


# Cohort Analysis Results
| Result Type | Primary Result | Others |
| :--  | :--      | :--   |
| Small_Variant | **<br>(1) "deepvariant.glnexus.phased.vcf.gz" <br>(2) "deepvariant.glnexus.phased.vcf.gz.tbi"** <br>* Small variants called by DeepVariant, joint-called by GLnexus, and phased by HiPhase (with index)| | |
| Structure_Variation | **<br>(1) "pbsv.phased.vcf.gz" <br>(2) "pbsv.phased.vcf.gz.tbi"** <br>* Structural variants joint-called by pbsv and phased by HiPhase (with index) |  |
| Haplotype_Phasing_Info | **<br>(1) "hiphase.blocks.tsv"** <br>* Phase block list written by HiPhase | **<br>(1) "hiphase.stats.tsv"** <br>* Phase block summary statistics written by HiPhase |


# Tertiary Analysis Results
| Result Type | Primary Result | Others |
| :--  | :--      | :--   |
| Small_Variant | **<br>(1) "deepvariant.glnexus.phased.norm.slivar.vcf.gz" <br>(2) "deepvariant.glnexus.phased.norm.slivar.vcf.gz.tbi"** <br>* Small variant calls that are filtered based on population frequency and annotated with cohort information, population frequency, gene, functional impact, etc. **<br>(1) "deepvariant.glnexus.phased.norm.slivar.compound_hets.vcf.gz" <br>(2)"deepvariant.glnexus.phased.norm.slivar.compound_hets.vcf.gz.tbi"** <br>* Compound heterozygotes annotated with cohort information, population frequency, gene, functional impact, etc.| **<br>(1) "deepvariant.glnexus.phased.norm.slivar.tsv" <br>(2) "deepvariant.glnexus.phased.norm.slivar.compound_hets.tsv"** <br>* Filtered VCFs are reformatted as a human-readable TSV by slivar tsv|
| Structure_Variation | **<br>(1) "pbsv.phased.svpack.vcf.gz" <br>(2)"pbsv.phased.svpack.vcf.gz.tbi"** <br>* Structural variant calls that are filtered based on population frequency and annotated with cohort information, population frequency, gene, functional impact, etc., using svpack | **<bar>(1) "pbsv.phased.svpack.tsv"** <br>* Filtered VCFs are reformatted as a human-readable TSV by slivar tsv|
